app/Services/Manager.php

```diff
@@ -92,6 +92,77 @@ public function exportCSV($model, $hasReportback = false)
     }
 
     /**
+     * Build leaderboard data for a given Competition
+     *
+     * @param Competition $competition
+     * @param User $users
+     * @return array $leaderboard
+     */
+    public function getLeaderboard($competition, $users)
+    {
+        $rows = [];
+
+        // Get all users in bulk
+        $parameters['ids'] = implode(',', $users->pluck('id')->all());
```

Comment:
Oh sorry. I didn't see that correctly. User the `UserRepository` with it's `getAll()` method, where you just pass an array of ids. In the code you reference, I use the Phoenix API cause for campaigns, they aren't cached currently. But user's should be via the Repository. If they're not cached, then the repository handles sending the call to Northstar. You're eschewing the use of the cache :wink: